xarray/core/dataset.py

```diff
@@ -9204,9 +9209,11 @@ def pad(
             (stat_length,) or int is a shortcut for before = after = statistic
             length for all axes.
             Default is ``None``, to use the entire axis.
-        constant_values : scalar, tuple or mapping of hashable to tuple, default: 0
-            Used in 'constant'.  The values to set the padded values for each
-            axis.
+        constant_values : scalar, tuple, mapping of hashable to tuple or
+            mapping of hashable to mapping of hashable to tuple, default: 0
```

Comment:
Not necessarily on this PR but do you happen to know why the default is listed as `0` here but is `None` in the func signature?

Reply:
Hum, what happened there, but it is a good point since if left at None the arrays are padded with `np.nan`. Should I change the default to "nan" ?

Reply:
Or to 0?

Reply:
Ah you mean changing the default and not the docstring?

Reply:
Ah sorry, so what's the actual value that goes onto the dataset, by default? I had originally thought it was `0`, but does `None` / `NaN` actually get placed on the dataset? (I realize it's listed as the default in the signature)

Reply:
I also wasn't sure when you first asked so I created a test for the default state and it does indeed look like it populates the padding with `np.nan` when `constant_values` is left at `None`. So we it is confusing to have `default: 0` in the doc string. We can either change the doc string to `None` or `NaN` or change the default in the function signature to `0` rather then `None`. I'm not sure which would be better, what do you think?

Reply:
Please change the docstrings, otherwise that's a breaking change!

Reply:
Great, thank you!